source/client/sound/SoundData.cpp

```diff
@@ -8,13 +8,35 @@
 
 #include "SoundData.hpp"
 
-// --------------------------------------------------------------------
-// WARNING! If you have an error here it is most likely because you did
-// not grab the sound data from a working Minecraft PE 0.1.3 .apk.
-// 
-// Check the readme for a guide on how to extract game sounds from the
-// Minecraft PE 0.1.3 .apk file.
-// --------------------------------------------------------------------
-#ifndef MISSING_SOUND_DATA
-#include "../../sound_data/sounds.h"
-#endif
+#include "client/app/AppPlatform.hpp"
+
+void SoundDesc::_load(const AppPlatform* platform, const char *name)
+{
+	if (isLoaded)
+	{
+		// Already Loaded
+		return;
+	}
+	// Read Asset File
+	AssetFile file = platform->readAssetFile(std::string("sound/") + name + ".pcm");
```

Comment:
I suggest replacing periods with slashes here. So it'll try to access, for example, `sound/step/gravel.pcm` instead of `sound/step.gravel.pcm`.

Reply:
There are no periods. It just accesses `gravel1.pcm`.

Reply:
Didn't catch that, sorry.  I find it kind of weird that there are multiple sounds with the same in-game ID, but it's probably just to support random sounds easily.